stock_portfolio/stock_portfolio/views/stock.py

```diff
@@ -54,12 +95,17 @@ def get_portfolio_symbol_view(request):
     try:
         stock = request.matchdict['symbol']
     except (IndexError, KeyError):
-        return HTTPNotFound()
+        raise HTTPNotFound()
 
     try:
         query = request.dbsession.query(Stock)
-        stock_detail = query.filter(Stock.symbol == stock)[0] # can also use .first() for first item
+        stock_detail = query.filter(Stock.account_id == request.authenticated_userid).filter(Stock.symbol == stock).one_or_none()
```

Comment:
This first `filter` does not work because `Stock.account_id` is a list of `Account` objects associated with a specific `Stock`.